nav2_bt_navigator/README.md

```diff
@@ -46,7 +46,21 @@ The BT Navigator package has two sample XML-based descriptions of BTs.  These tr
 </root>
 ```
 
-Navigate with replanning is composed of the following custom decorator, condition and action nodes:
+Navigate with replanning is composed of the following custom decorator, condition,
+contro and action nodes:
+
+#### Control Nodes
+* PipelineSequence: This is a variant of a Sequence Node. When this node is ticked,
+it will tick the first child till it succeeds. Then it will tick the first two
+children till the second one succeeds. Then it will tick the first three till the
+third succeeds and so on, till there are no more children. This will return RUNNING,
+till the last child succeeds, at which time it also returns SUCCESS. If any child
+returns FAILURE, all nodes are halted and this node returns FAILURE.
+* Recovery: This is a control flow type node with two children.  It returns success if and only if the first child returns success. The second child will be executed only if the first child returns failure.  The second child is responsible for recovery actions such as re-initializing system or other recovery behaviors. If the recovery behaviors are succeeded, then the first child will be executed again.  The user can specify how many times the recovery actions should be taken before returning failure. The figure below depicts a simple recovery node.
```

Comment:
Can one BT node talk to another? 

Conceptually, we could add a method to the recoveries (or a BT node between the recoveries) to check if the controller/planner nodes can now compute a valid effort. This is for the global recovery/recovery state.

Reply:
Yes. Sort of. One node can output some data to a port, and another node can read in from that port.

For example, ComputePathToPose writes the path, and FollowPath reads that path.

Is that the kind of talking you are referring to?

Reply:
No, I’m asking if a node can see the current status of another

Reply:
No. Not without changing the code. Somehow, the status would need to be posted to an output port.

@mhpanah and I were talking about that as an alternate way of approaching this problem. We could have a node directly post its status. We also talked about creating a decorator node that posts its child status to a port. That way you could grab the status anywhere in the tree.

It feels a bit kludgy doing that, but, given the limitations of existing behavior tree nodes, it might be appropriate in certain circumstances.

Reply:
What’s the overhead of a port? We could also just make a ros topic that nodes publish to when they start and terminate. Have the publisher be a blackboard param like the node pointer. Lets outside people also listen in for metrics or state of BT

Reply:
Either way this doesnt seem aimed at the recoveries stuff so this comment is moot

Reply:
> What’s the overhead of a port?

The blackboard is basically a std::map. The overhead is on the order of copying into or out of the map.